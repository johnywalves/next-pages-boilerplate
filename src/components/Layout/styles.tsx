import styled from 'styled-components'

export const WrapperLayout = styled.main`
  background-color: var(--color-background);
  color: var(--color-text);
  width: 100%;
  min-height: 100%;
  padding: var(--30px);
  text-align: center;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: var(--16px);

  & button {
    background-color: transparent;
    color: var(--color-text);
    border: 1px solid var(--color-primary);
    padding: var(--16px) var(--24px);
    font-size: 1.25rem;
    font-weight: 600;
    cursor: pointer;
    transition: background-color 0.25s;

    &:hover {
      background-color: var(--color-primary);
      color: var(--color-text);
    }
  }
`
