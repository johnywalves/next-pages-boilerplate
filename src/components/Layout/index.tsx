import React, { ReactNode } from 'react'

import { WrapperLayout } from './styles'

type LayoutProps = {
  children: ReactNode
}

const Layout = ({ children }: LayoutProps) => (
  <WrapperLayout>{children}</WrapperLayout>
)

export default Layout
