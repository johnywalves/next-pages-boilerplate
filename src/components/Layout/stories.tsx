import { Story, Meta } from '@storybook/react/types-6-0'

import Layout from '.'

export default {
  title: 'Layout',
  component: Layout
} as Meta

const Template: Story = ({ children, ...rest }) => (
  <Layout {...rest}>{children}</Layout>
)

export const Basic = Template.bind({})
