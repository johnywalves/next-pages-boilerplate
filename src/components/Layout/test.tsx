import { render, screen } from '@testing-library/react'

import Layout from '.'

describe('<Layout />', () => {
  it('should render the heading', () => {
    const { container } = render(
      <Layout>
        <h1>Layout</h1>
      </Layout>
    )

    expect(screen.getByRole('heading', { name: /layout/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
