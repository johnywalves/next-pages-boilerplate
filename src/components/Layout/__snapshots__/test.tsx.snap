// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Layout /> should render the heading 1`] = `
.c0 {
  background-color: var(--color-background);
  color: var(--color-text);
  width: 100%;
  min-height: 100%;
  padding: var(--30px);
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: var(--16px);
}

.c0 button {
  background-color: transparent;
  color: var(--color-text);
  border: 1px solid var(--color-primary);
  padding: var(--16px) var(--24px);
  font-size: 1.25rem;
  font-weight: 600;
  cursor: pointer;
  -webkit-transition: background-color 0.25s;
  transition: background-color 0.25s;
}

.c0 button:hover {
  background-color: var(--color-primary);
  color: var(--color-text);
}

<main
  class="c0"
>
  <h1>
    Layout
  </h1>
</main>
`;
